pytket/pytket/rmbqc/utils.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2019-2022 Cambridge Quantum Computing
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from pytket.circuit import Command, Circuit  # type: ignore
+
+
+def is_Clifford(aGate: Command) -> bool:
+    """
+    Check if a gate is Clifford.
+    
+    :param aGate:    Command to check.
+    :param type:     Command
+    
+    :returns:        The result of the check.
+    :rtype:          bool
+    """
+    if aGate.op.get_name() in {
+        "Z",
+        "X",
+        "Y",
+        "H",
+        "S",
+        "V",
+        "Sdg",
+        "Vdg",
+        "SX",
+        "SXdg",
+        "CX",
+        "CY",
+        "CZ",
+        "CH",
+        "CV",
+        "CVdg",
+        "CSX",
+        "CSXdg",
+        "CCX",
+        "SWAP",
+        "CSWAP",
+        "noop",
+        "BRIDGE",
+        "Reset",
+    }:
+        return True
+    elif aGate.op.get_name() in {"T", "Tdg"}:
+        return False
+    elif aGate.op.get_name() in {"Rx", "Rz", "Ry", "CRx", "CRy", "CRz"}:
+        if aGate.op.params[0] in {0, 1 / 2, 1, 3 / 2, 2}:
+            return True
+    return False
+
+
+def count_nCliffords(c: Circuit) -> int:
+    """
+    Returns number of non-Clifford gates in a circuit.
+    
+    :param c:        Circuit to check.
+    :param type:     Circuit
+    
+    :returns:        The number of non-Clifford gates.
+    :rtype:          int
+    """
+    count = 0
+    for g in c.get_commands():
+        if is_Clifford(g):
+            count += 1
+    return count
+
+
+def is_worthwhile(
+    self,
+    improveOn: str = "Depth",
```

Comment:
done, switched to a boolean since we never decrease width in mbqc anyway... so the question is whether we only care about depth or we care about both width and depth.